rustls/src/verify.rs

```diff
@@ -458,7 +458,7 @@ impl UnparsedCertRevocationList {
 /// A `ClientCertVerifier` that will ensure that every client provides a trusted
 /// certificate, without any name checking. Optionally, client certificates will
 /// have their revocation status checked using the DER encoded CRLs provided.
-pub struct AllowAnyAuthenticatedClient {
+pub(crate) struct AllowAnyAuthenticatedClient {
```

Comment:
Now this is non-public, maybe we could rename it, get rid of `AllowAnyAnonymousOrAuthenticatedClient`, and just allow configuration of `client_auth_mandatory`'s return?

Reply:
> Now this is non-public, maybe we could rename it

Maybe I should smoosh the `WebpkiClientVerifier` that I added in `verifier_builder.rs` as an empty type that's only good for producing a `ClientCertVerifierBuilder` w/ `builder()` into this type, keeping the new name and having it be the impl of the `ClientCertVerifier` trait. It felt a bit awkard having that `WebpkiClientVerifier` type that wasn't actually a verifier, just a gateway to a builder.

>  get rid of AllowAnyAnonymousOrAuthenticatedClient, and just allow configuration of client_auth_mandatory's return?

Makes sense to me :+1:

Reply:
Done :stars: